### PR TITLE
fix: fix query failure on simultaneous execution of multiple queries

### LINKED ISSUE #15333: [Bug]: Redshift multiple actions failing on page load
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

https://theappsmith.slack.com/archives/C03K6MJMG04/p1658223617868939

### Steps To Reproduce

https://theappsmith.slack.com/archives/C03K6MJMG04/p1658223617868939

### Public Sample App

_No response_

### Version

self hosted, cloud

### PR DESCRIPTION
**Pull request by sumitsum**
## Description
- The default connection that Redshift driver provides is not thread safe. This leads to query failure on simultaneous execution of multiple queries. This PR replaces the connection object by a connection pool.
- Update the Redshift driver version.
- Hide away SSL options from the datasource config page since they are not handled at the moment.

Fixes #15333 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Manual
- JUnit TC
- Not adding a cypress test because AFAIK, we don't have a Redshift DB instance in our CI test setup. 

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/utils/RedshiftDatasourceUtils.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        }
+
+        // Set up the connection URL
+        StringBuilder urlBuilder = new StringBuilder(JDBC_PROTOCOL);
+
+        List<String> hosts = datasourceConfiguration
+                .getEndpoints()
+                .stream()
+                .map(endpoint -> endpoint.getHost() + ":" + ObjectUtils.defaultIfNull(endpoint.getPort(), 5432L))
+                .collect(Collectors.toList());
```
Comment (subrata71) [inline app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/utils/RedshiftDatasourceUtils.java:52]: @sumitsum 
**5432** is the default port for Postgres. Can you please check again?

Comment (sumitsum) [reply app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/utils/RedshiftDatasourceUtils.java:52]: @subrata71 nic catch ! I have updated the default port to 5439.

**Review by subrata71 — DISMISSED** [review]
Looks good to me

**Comment by sumitsum** [comment]
Not adding cypress test since we don't have Redshift test DB in CI. cc: @prapullac

**Review by nidhi-nair — APPROVED** [review]

**Comment by nidhi-nair** [comment]
Merging in since tests have passed but status has not been updated.

**Pull request merged**